dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/BasicHttpAuthenticationMechanism.java

```diff
@@ -39,24 +41,40 @@
 import com.ibm.websphere.ras.annotation.Sensitive;
 import com.ibm.ws.common.internal.encoder.Base64Coder;
 import com.ibm.ws.security.authentication.AuthenticationConstants;
+import com.ibm.ws.security.javaeesec.authentication.mechanism.http.HAMProperties;
+import com.ibm.ws.security.javaeesec.JavaEESecConstants;
 import com.ibm.wsspi.security.token.AttributeNameConstants;
 
 @Default
 @ApplicationScoped
 public class BasicHttpAuthenticationMechanism implements HttpAuthenticationMechanism {
+    @Inject
+    Instance<HAMProperties> hampInstance;
 
     private static final TraceComponent tc = Tr.register(BasicHttpAuthenticationMechanism.class);
 
-    private final String realmName;
+    private String realmName = null;
     private final String DEFAULT_REALM = "defaultRealm";
 
     /**
-     * @param realmName
+     *
      */
-    public BasicHttpAuthenticationMechanism(String realmName) {
-        this.realmName = realmName;
+    public BasicHttpAuthenticationMechanism() {
+        if (hampInstance != null && !hampInstance.isUnsatisfied() && !hampInstance.isAmbiguous()) {
+            HAMProperties hamp = hampInstance.get();
+            if (hamp != null) {
+                Properties props = hamp.getProperties();
+                if (props != null) {
+                    realmName = (String)props.get(JavaEESecConstants.REALM_NAME);
+                }
+            }
+        }
+        if (realmName == null) {
+            Tr.warning(tc, "JAVAEESEC_WARNING_NO_REALM_NAME");
```

Comment:
Should check for the trace guards.

Reply:
This would be IdenttyStoreHandler. Let me fix it.